lib/src/main/java/com/orange/ods/compose/component/navigationdrawer/OdsModalDrawer.kt

```diff
@@ -0,0 +1,253 @@
+    content: @Composable () -> Unit,
+    headerParametersProvider: OdsModalDrawerHeaderParametersProvider,
+    listContent: List<OdsModalDrawerItem>
+) {
+    ModalDrawer(
+        drawerContent = {
+            OdsModalDrawerHeader(headerContent = headerParametersProvider)
+            OdsDivider()
+            listContent.forEach { item ->
+                getItem(item = item).invoke()
+            }
+        },
+        drawerState = drawerState,
+        modifier = modifier,
+        drawerBackgroundColor = OdsTheme.colors.surface,
+        drawerContentColor = contentColorFor(OdsTheme.colors.onSurface),
+        scrimColor = DrawerDefaults.scrimColor,
+        content = content
+    )
+}
+
+sealed class OdsModalDrawerItem
+class OdsModalDrawerSectionLabel(val label: String) : OdsModalDrawerItem()
+object OdsModalDrawerDivider : OdsModalDrawerItem()
+class OdsModalDrawerListItem(@DrawableRes val icon: Int?, val text: String) : OdsModalDrawerItem()
+
+internal fun getItem(item: OdsModalDrawerItem): @Composable (() -> Unit) {
+    return when (item) {
+        is OdsModalDrawerSectionLabel -> {
+            {
+                Column {
+                    OdsDivider()
+                    OdsTextSubtitle2(
+                        modifier = Modifier.padding(top = dimensionResource(id = R.dimen.spacing_m), start = dimensionResource(id = R.dimen.spacing_m)),
+                        text = item.label
+                    )
+                }
+            }
+        }
+        is OdsModalDrawerListItem -> {
+            {
+                OdsListItem(
+                    modifier = Modifier
+                        .iconType(OdsListItemIconType.Icon)
+                        .background(color = Color.Red),
+                    text = item.text,
+                    icon = item.icon?.let { { OdsListItemIcon(painterResource(id = it)) } })
+            }
+        }
+        is OdsModalDrawerDivider -> {
+            {
+                OdsDivider()
+            }
+        }
+    }
+}
+
+data class OdsModalDrawerHeaderParametersProvider(
+    var modifier: Modifier = Modifier,
+    var title: String,
+    var imageContentDescription: String? = null,
+    var backgroundImage: Painter? = null,
+    var subtitle: String? = null,
+    var avatar: Painter? = null
+)
+
+@Composable
+internal fun OdsModalDrawerHeader(
+    headerContent: OdsModalDrawerHeaderParametersProvider
+) {
+    Box(
+        modifier = headerContent.modifier
+            .fillMaxWidth()
+            .height(
+                if (headerContent.backgroundImage != null || headerContent.avatar != null) 167.dp else if (headerContent.subtitle != null) dimensionResource(
+                    id = R.dimen.list_two_line_item_height
+                ) else dimensionResource(id = R.dimen.list_single_line_item_height)
+            ),
+    ) {
+        headerContent.backgroundImage?.let {
+            Image(
+                painter = it,
```

Comment:
Always give a name to the `it` in order to have a more readable code except if this is a small expression. It can be `backgroundPainter`.